File: src/app/class/contrat.ts
import { Assure } from "./assure";

export class Contrat {

    id!: number;
    police: string;
    dateSignature: Date;
    dateExpiration: Date;
    
    assure: Assure;
  
    constructor() {
      this.police = '';
      this.dateSignature = new Date();
      this.dateExpiration = new Date();
      
      this.assure =new Assure();
    }
}

File: assurance/src/main/java/com/tarmiz/assurance/model/Contrat.java
package com.tarmiz.assurance.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "contrat")
public class Contrat {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String police;
    private LocalDate dateSignature;
    private LocalDate dateExpiration;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "assure_id")
    private Assure assure;

}
